src/sentinel.c

```diff
@@ -4317,15 +4319,25 @@ void sentinelSetCommand(client *c) {
         }
     }
 
-    if (changes) sentinelFlushConfig();
+    if (changes) {
+        if (sentinelFlushConfig() == C_ERR) {
+            addReplyErrorFormat(c,"Failed to save Sentinel new configuration on disk");
+            return;
+        }
+    }
     addReply(c,shared.ok);
     return;
 
 badfmt: /* Bad format errors */
     addReplyErrorFormat(c,"Invalid argument '%s' for SENTINEL SET '%s'",
         (char*)c->argv[badarg]->ptr,option);
 seterr:
-    if (changes) sentinelFlushConfig();
+    if (changes) {
+        if (sentinelFlushConfig() == C_ERR) {
+            addReplyErrorFormat(c,"Failed to save Sentinel new configuration on disk");
+            return;
+        }
+    }
```

Comment:
please combine two if statements into one.

Reply:
Yes, you are right. They can be merged to one statement.  Thanks